.github/workflows/application-signals-e2e-test.yml

```diff
@@ -27,21 +37,23 @@ jobs:
       caller-workflow-name: 'main-build'
 
   java-ec2-default-e2e-test:
+    needs: CheckBuildTestArtifacts
     uses: aws-observability/aws-application-signals-test-framework/.github/workflows/java-ec2-default-e2e-test.yml@main
     secrets: inherit
     with:
       aws-region: us-east-1
       caller-workflow-name: 'main-build'
 
   java-ec2-asg-e2e-test:
+    needs: [ CheckBuildTestArtifacts ]
```

Comment:
nit: why is it a list if there is only one dependency

Reply:
They did that for their other jobs, so I just followed the pattern.